packages/cactus-plugin-ledger-connector-iroha/src/test/typescript/integration/run-transaction-endpoint-v1.test.ts

```diff
@@ -0,0 +1,786 @@
+  const server = http.createServer(expressApp);
+  const listenOptions: IListenOptions = {
+    hostname: "0.0.0.0",
+    port: 0,
+    server,
+  };
+  const addressInfo = (await Servers.listen(listenOptions)) as AddressInfo;
+  test.onFinish(async () => await Servers.shutdown(server));
+  const { address, port } = addressInfo;
+  const apiHost = `http://${address}:${port}`;
+  const apiConfig = new Configuration({ basePath: apiHost });
+  const apiClient = new IrohaApi(apiConfig);
+
+  await connector.getOrCreateWebServices();
+  await connector.registerWebServices(expressApp);
+
+  let firstTxHash;
+  const adminAccount = iroha.getGenesisAdminAccount();
+  {
+    const req = {
+      commandName: IrohaCommand.CreateAccount,
+      params: [
+        "user1",
+        "test",
+        "fe31d7bc2dbe695b1cd5c706dde902d5fb6e6ce4b60e08418c842fdef5805230", //node4.pub
+      ],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.equal(res.data.transactionReceipt.status, "COMMITTED");
+  }
+
+  {
+    const req = {
+      commandName: IrohaQuery.GetAccount,
+      params: [adminAccount],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    console.log(res.data.transactionReceipt);
+    t.deepEqual(res.data.transactionReceipt, {
+      accountId: adminAccount,
+      domainId: "test",
+      quorum: 1,
+      jsonData: "{}",
+    });
+  }
+
+  {
+    const req = {
+      commandName: IrohaCommand.CreateDomain,
+      params: ["test2", "admin"],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.equal(res.data.transactionReceipt.status, "COMMITTED");
+  }
+
+  {
+    const req = {
+      commandName: IrohaCommand.CreateAsset,
+      params: ["coolcoin", "test", 3],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.equal(res.data.transactionReceipt.status, "COMMITTED");
+  }
+
+  {
+    const req = {
+      commandName: IrohaQuery.GetAssetInfo,
+      params: ["coolcoin#test"],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.deepEqual(res.data.transactionReceipt, {
+      assetId: "coolcoin#test",
+      domainId: "test",
+      precision: 3,
+    });
+  }
+
+  {
+    const req = {
+      commandName: IrohaCommand.AddAssetQuantity,
+      params: ["coolcoin#test", "123.123"],
+    };
+    const res = await apiClient.runTransactionV1(req as RunTransactionRequest);
+    t.ok(res);
+    t.ok(res.data);
+    t.equal(res.status, 200);
+    t.equal(res.data.transactionReceipt.status, "COMMITTED");
```

Comment:
I see the code multiple times, woudn't it be better if You extract function?

Reply:
We should discuss about it during meeting.